### PR TITLE
[pyspark] Add param validation for "objective" and "eval_metric" param, and remove invalid booster params

### PR DESCRIPTION
**Pull request by WeichenXu123**
* Added param validation for "objective" and "eval_metric" param
* Removed invalid booster params passed to `xgboost.train`
* Allow regressor setting other objective value
* Handle the case running on spark GPU cluster but `spark.task.resource.gpu.amount` config is not set.

### CODE REVIEW

**Comment by WeichenXu123** [comment]
CC @trivialfis @wbo4958

**Review by trivialfis — APPROVED** [review]

**Comment by trivialfis** [comment]
Please fix the formatting error: https://github.com/dmlc/xgboost/runs/7824147126?check_suite_focus=true .

File: python-package/xgboost/spark/core.py [hunk 1]
```diff
@@ -267,6 +267,18 @@ def _validate_params(self):
                 "If features_cols param set, then features_col param is ignored."
             )
 
+        if self.getOrDefault(self.objective) is not None:
```
Comment (wbo4958) [inline python-package/xgboost/spark/core.py:270]: It seems not make sense (to me) to check the type of a parameter that already has the typeConverter

Comment (WeichenXu123) [reply python-package/xgboost/spark/core.py:270]: The "objective" param is automatically generated, currently it does not have type converter.

File: python-package/xgboost/spark/core.py [hunk 2]
```diff
@@ -345,7 +353,14 @@ def _validate_params(self):
                 )
             else:
                 # checking spark non-local mode.
-                if not gpu_per_task or int(gpu_per_task) < 1:
+                if not gpu_per_task:
+                    # For spark cluster with GPU config, the "spark.task.resource.gpu.amount"
+                    # config might not be set. In this case, the default gpu per task is 1.
+                    # If running on no GPU available, in spark task calling `_get_gpu_id` will
+                    # raise gpu not found error.
+                    gpu_per_task = "1"
```
Comment (WeichenXu123) [inline python-package/xgboost/spark/core.py:361]: @wbo4958 A notable fix.

**Comment by WeichenXu123** [comment]
CC @trivialfis @wbo4958 Done.

File: python-package/xgboost/spark/core.py (hunk 2 above)
Comment (WeichenXu123) [reply python-package/xgboost/spark/core.py]: My fault, reverted this.

File: python-package/xgboost/spark/core.py (hunk 1 above)
Comment (wbo4958) [reply python-package/xgboost/spark/core.py:290]: I still hope we can find a common solution for the type converter in the following PR

**Review by wbo4958 — APPROVED** [review]

**Comment by WeichenXu123** [comment]
@trivialfis @wbo4958 We can merge this. The test failures are not related to my PR.

**Pull request merged**